rules-java/api/src/main/resources/reports/templates/java_application.ftl

```diff
@@ -64,11 +64,18 @@
             <h4 class="warns">Warnings: ${warnings} items</h4>
             <#if warnings != 0 >
                 <ul class='notifications'>
+                    <#assign map = {}>
                     <#list sourceReportModel.sourceFileModel.classificationModels.iterator() as classification>
-                        <li class='warning clsf'>${classification.classification?html}</li>
+                        <#assign count = (map[classification.classification]!0) + 1>
+                        <#assign map += {classification.classification : count}>
                     </#list>
                     <#list sourceReportModel.sourceFileModel.inlineHints.iterator() as hintLine>
-                        <li class='warning hint'>${hintLine.title?html}</li>
+                        <#assign count = (map[hintLine.title]!0) - 1>
+                        <#assign map += {hintLine.title : count}>
```

Comment:
If there are same hint lines as classification titles, this trick would not work well... Can redo if you think it's likely.